src/main/java/org/fenixedu/academic/domain/degreeStructure/ProgramConclusion.java

```diff
@@ -196,6 +196,9 @@ protected void checkForDeletionBlockers(Collection<String> blockers) {
 
     public void delete() {
         DomainException.throwWhenDeleteBlocked(getDeletionBlockers());
+        if (!getProgramConclusionConfigsSet().isEmpty()) {
```

Comment:
If you see more carefully, this delete method have a mechanism to check deletion "blockers": conditions to prevent the deletion like having program conculsion configs. In order to respect the consistency, this check should be in the  proper method: `checkForDeletionBlockers(..)`